OfflineService/OfflineServiceViewController.swift

```diff
@@ -0,0 +1,300 @@
+    
+    var offlineDataItems = [OfflineDataItem]()
+    let tilesUnpackingLock = NSLock()
+    
+    // MARK: - OfflineServiceObserver methods
+    
+    public func onPending(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function), identifier: \(metadata.id)")
+    }
+    
+    public func onDownloading(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function), identifier: \(metadata.id), progress: \(pack.bytes) bytes")
+        
+        for index in (0 ..< offlineDataItems.count) {
+            if offlineDataItems[index].dataRegionMetadata.id != metadata.id { continue }
+            
+            DispatchQueue.main.async {
+                let indexPath = IndexPath(row: index, section: 0)
+                let cell = self.tableView.cellForRow(at: indexPath) as? OfflineDataRegionTableViewCell
+                
+                cell?.showDownloadProgress(for: domain, dataPack: pack, metadata: metadata)
+            }
+        }
+    }
+    
+    public func onIncomplete(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function), identifier: \(metadata.id)")
+    }
+    
+    public func onVerifying(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function), identifier: \(metadata.id)")
+    }
+    
+    public func onAvailable(for domain: OfflineDataDomain, metadata: OfflineDataRegionMetadata, pack: OfflineDataPack) {
+        print("[OfflineServiceObserver] \(#function), identifier: \(metadata.id)")
+
+        updateOfflineDataRegions(for: domain, metadata: metadata)
+        
+        switch domain {
+        case .maps:
+            // Maps SDK will automatically pick up downloaded offline pack
+            break
+        case .navigation:
+            tilesUnpackingLock.lock()
+            
+            do {
+                guard let outputDirectoryURL = Bundle.mapboxCoreNavigation.suggestedTileURL?.appendingPathComponent("unpacked") else { return }
+                guard let packName = URL(string: pack.path)?.lastPathComponent else { return }
+                guard let packData = FileManager.default.contents(atPath: pack.path) else { return }
+                
+                try FileManager.default.createDirectory(at: outputDirectoryURL, withIntermediateDirectories: true, attributes: nil)
+                let temporaryPackURL = outputDirectoryURL.appendingPathComponent(packName).appendingPathExtension("tar")
+                try packData.write(to: temporaryPackURL)
+                
+                NavigationDirections.unpackTilePack(at: temporaryPackURL, outputDirectoryURL: outputDirectoryURL, progressHandler: { (totalBytes, unpackedBytes) in
```

Comment:
I have some concerns regarding this API, since we have to unpack archive so it can be consumed by Navigation Native all possible regions are unpacked in the same folder. For example here's approximate structure for several regions:
<img width="1038" alt="Screen Shot 2020-10-09 at 3 56 46 PM" src="https://user-images.githubusercontent.com/1496498/95638008-2df6bc80-0a48-11eb-931a-e9bc8fd555d7.png">

As you can see due to folder naming in unarchived folders we might get conflicts which might lead to overwrite of certain files. There is also another problem with removal of Navigation offline region when using OfflineService API: even though original archive will be removed we no longer have control over unpacked files, thus they won't be removed and user will still be able to use them. This problem is partly fixed by calling `removeUnpackedTilesDirectory()` each time app starts - it'll remove whole unpacked folder and will unpack all available regions once again.

/cc @1ec5, @IodaMikeMapbox, @SiarheiFedartsou

Reply:
No longer relevant.